pkg/query/endpointset.go

```diff
@@ -302,39 +306,44 @@ func (e *EndpointSet) Update(ctx context.Context) {
 	e.updateMtx.Lock()
 	defer e.updateMtx.Unlock()
 
-	e.endpointsMtx.RLock()
-	endpoints := make(map[string]*endpointRef, len(e.endpoints))
-	for addr, er := range e.endpoints {
-		endpoints[addr] = er
-	}
-	e.endpointsMtx.RUnlock()
-
-	level.Debug(e.logger).Log("msg", "starting to update API endpoints", "cachedEndpoints", len(endpoints))
-
-	activeEndpoints := e.getActiveEndpoints(ctx, endpoints)
-	level.Debug(e.logger).Log("msg", "checked requested endpoints", "activeEndpoints", len(activeEndpoints), "cachedEndpoints", len(endpoints))
+	currentEndpoints := e.endpointRefs()
```

Comment:
```suggestion
	currentEndpoints := e.cachedEndpoints()
```
I am missing what kind of refs and endpoints we are returning here